app/src/main/java/org/kiwix/kiwixmobile/utils/LanguageUtils.java

```diff
@@ -341,16 +341,20 @@ public String getLanguageName() {
     }
   }
 
-  public static String getStringOrResource(Context appContext, String str){
-    if(str.contains("REPLACE_")) {
-      int resourceId = appContext.getResources()
-              .getIdentifier(
-                      str.replace("REPLACE_", ""),
-                      "string",
-                      appContext.getPackageName()
-              );
-      return appContext.getResources().getString(resourceId);
+
+
+  public static String getResourceString(Context appContext, String str){
+    String resourceName = str;
+    if(resourceName.contains("REPLACE_")) {
+      resourceName = resourceName.replace("REPLACE_", "");
     }
-    return str;
+    int resourceId = appContext.getResources()
+      .getIdentifier(
+        resourceName,
+        "string",
+        appContext.getPackageName()
+      );
```

Comment:
Some tests would be cool to show what happens when a match is found and when a match isn't found.

Reply:
It's difficult to test as the tests don't have access to the App Context, do you have any ideas RE that?